src/plugins/path/README.md

```diff
@@ -5,28 +5,75 @@
+as correct permissions
 
 ## Introduction
 
-This plugin checks whether the value of a key is a valid file system path. 
+This plugin checks whether the value of a key is a valid file system path and optionally if
+correct permissions are set for a certain user.
 
 ## Purpose
 
 The motivation to write this plugin is given by the two paths that exist
-in /etc/fstab: the device file and the mount point. A missing file is
+in /etc/fstab: the device file and the mountpoint. A missing file is
 not necessarily an error, because the device file may appear later when
-a device is plugged in and the mount point may be there when another
+a device is plugged in and the mountpoint may be there when another
 subsequent mount was executed. So only warnings are yielded in that
 case. One situation, however, presents an error: Only an absolute path
-is allowed to occur for both device and mount point. When checking for
+is allowed to occur for both device and mountpoint. When checking for
 relative files, it is not enough to look at the first character if it is
 a `/`, because remote file systems and some special names are valid, too.
 
+If `check/path/mode = [permission]` is also present it will check for the correct permissions
+of the file/directory. Optionally, you can also add `check/path/user = [user]"` which then checks the permissions
+for the given user. When calling `kdb set` on the actual key, you have to run as `root` user
+or the file permissions cannot be checked (you will receive an error message). It is also possible to leave the
+`check/path/user` empty (just provide an empty string) which then takes the executing user as target to check.
+So for example `sudo kdb set ...` will check if `root` can access the target file/directory whereas `kdb set ...`
+will take the current executing process/user. If `check/path/user` is not given at all, the plugin
+will check accessibility for the `root` user only (which again requires `sudo`)
+
+ `check/path/mode = rw` and `check/path/user = tomcat` for example will check if the user
+`tomcat` has read and write access to the path which was set for the key. Please note that the file has to exist already
+and it is not checked if the user has the right to create a file in the directory.
+
+ Permissions available:
+ - `r`: **R**ead
+ - `w`: **W**rite
+ - `x`: e**X**ecute
+
 ## Usage
 
 If the metakey `check/path` is present, it is checked if the value is a
 valid absolute file system path. If a metavalue is present, an additional
 check will be done if it is a directory or device file.
 
+## Example
+
+```sh
+sudo kdb mount test.dump /test path dump
+sudo kdb setmeta /test/path check/path ""
+#> Using keyname spec/test/path
+sudo kdb setmeta /test/path check/path/user ""
+#> Using keyname spec/test/path
+sudo kdb setmeta /test/path check/path/mode "rw"
+#> Using keyname spec/test/path
+
+# Standard users should not be able to read/write the root folder
+kdb set /test/path "/root"
+# ERROR:210
```

Comment:
What Do you mean by "the Mode is not correct"?

Reply:
TMPFILE=$(mktemp)
check/path/mode rwx
chmod +rw $TMPFILE
kdb set ... $TMPFILE  # should fail because rwx is required

Reply:
Unfortunately shell variables are not being saved:
`CMD: export TMPFILE=$(mktemp)

RET: 0



CMD: chmod +rw "$TMPFILE"

RET: 1

=== FAILED return value does not match expected pattern 0

STDERR: chmod: cannot access '': No such file or directory
`

Reply:
I could make up my own file in the directory with a long random hardcoded string if this is acceptable.

Reply:
You can use `kdb` to store temporary data:

```sh
# Create a temporary file and store the filepath in `user/tests/path/tempfile`
kdb set user/tests/path/tempfile $(mktemp)
# Retrieve path of the temporary file
kdb get user/tests/path/tempfile
```

.

Reply:
Very creative way @sanssecours :+1: !
I tried it with the hardcoded way now and it works even though `touch` did not work interestingly.

Since it is an example I think it looks nicer for users than something like this:
"kdb set `kdb get user/tests/path/tempfile`". Don't you think?

Reply:
While it looks nicer, hardcoded file paths will be a problem, if multiple instances of a Markdown Shell Recorder test access the same file concurrently. This was for example the case on the old Jenkins build server infrastructure, where files were shared between concurrent build jobs that ran on the same machine. While I assume the new Docker infrastructure should have fixed this problem, it is always nice to have tests that work in parallel too.